cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -79,20 +83,25 @@ public Path getBasePath() {
   public static void validateConfig(SecureStore secureStore, SourceControlConfig sourceControlConfig) throws
     RepositoryConfigValidationException {
     RepositoryConfig config = sourceControlConfig.getRepositoryConfig();
+    CredentialsProvider credentialsProvider = sourceControlConfig.getAuthenticationStrategy()
+      .getCredentialsProvider(secureStore, config, sourceControlConfig.getNamespaceID());
     // Try fetching heads in the remote repository.
     try (Git git = Git.wrap(new InMemoryRepository.Builder().build())) {
-      LsRemoteCommand cmd = createCommand(git::lsRemote, sourceControlConfig, secureStore).setRemote(config.getLink())
-                                                                                          .setHeads(true)
-                                                                                          .setTags(false);
+      LsRemoteCommand cmd =
+        createCommand(git::lsRemote, sourceControlConfig, credentialsProvider).setRemote(config.getLink())
+          .setHeads(true)
+          .setTags(false);
       validateDefaultBranch(cmd.callAsMap(), config.getDefaultBranch());
     } catch (TransportException e) {
       throw new RepositoryConfigValidationException(
         "Failed to connect with repository. Authentication credentials invalid or the repository may not exist.",
         e);
     } catch (RepositoryConfigValidationException e) {
       throw e;
-    } catch (AuthenticationConfigException e) {
+    } catch (IllegalStateException e) {
       throw new RepositoryConfigValidationException(e.getMessage(), e);
+    } catch (GitAPIException e) {
+      throw new RepositoryConfigValidationException("Failed to list remotes: " + e.getMessage(), e);
```

Comment:
whats the message be more elaborate? How it it different from L106?

Reply:
There is no reason for the difference. Made them the same.